sdk/cosmos/azure-cosmos/docs/StatusCodes.md

```diff
@@ -0,0 +1,53 @@
+|404|1024|x|`Not Found/Incorrect Container resource id` - Indicates that a client attempted to use a container that has recently been deleted and rectreated. So, the cached container id in the client is stale - and identifies the previosuly deleted container. The SDK will trigger retries - in general applications need to be able to tolerate that conatiner deletion and immediate recreation will take up-to a few seconds/minutes to be replicated across all regions.|
+|408|\*|x|`Request timeout` - Indicates a tiemout for an attempted operation. See [TSG](https://learn.microsoft.com/azure/cosmos-db/nosql/troubleshoot-java-sdk-request-timeout) for more details.|
+|408|20008|(x)|`Request timeout/End-to-end timeout exceeded` - Indicates that the application defined end-to-end timeout was exceeded when processing an operation. This will usually be a transient error condition - exceptions are when the application defines unrealistis end-to-end timeouts - for example when executing a query that could very well take a few seconds because it is relatively inefficient or when the end-to-end timeout is lower than the to-be-expected network transit time between the application's location and the Cosmso DB sevice endpoint.|
+|408|20901||`Request timeout/Negative End-to-end timeout provided` - Indicates that the application defined a negative end-to-end timeout. This indicates a bug in your application.|
+|409|0||`Conflict` - Indicates that the attempt to insert (or upsert) a new document cannot be processed because another document with the same identity (partition key value + value of `id` property) exists or a unqiue key constraint would be violated.|
+|410|\*|x|`Gone` - indicates transient error conditions that could happen while replica get moved to a different node or partitions get split/merged. The SDK will retry these error conditions and usually mitigate them without even surfacing them to the application. If these erros get surfaced to the application as `CosmosException` with status code `410` or `503` thses errors should always be transient.|
+|410|21010|x|`Service timeout` - Indicates that an operation has been timed out at the service.  See [TSG](https://learn.microsoft.com/azure/cosmos-db/nosql/troubleshoot-request-timeout) for more details. This error will be mapped to a CosmosException with status code `503` when surfacing it to the application after exceeding SDK-retries.|
+|410|21001|x|`Name cache is stale` - Indicates that a container was deleted and recreated - and the client's cache still has the old container metadata. This error indicates that the client even after refreshing the cache got the container metadata of the "old" container. Usually it indicates that the replication of the new container metadata across all regions took longer than usual. This error should always be transient. This error will be mapped to a CosmosException with status code `503` when surfacing it to the application after exceeding SDK-retries.|
```

Comment:
they should never see 410/21001 -> we only use this substrates code when bubbling up as a 503. So I wonder we should put into the 503/21001 section

Reply:
Thanks fixed in next iteration